providers/grafana-on-call/src/lib/grafana-on-call.provider.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IChatOptions,
+  IChatProvider,
+} from '@novu/stateless';
+import axios from 'axios';
+
+export class GrafanaOnCallChatProvider implements IChatProvider {
+  id = 'grafana-on-call';
+  channelType = ChannelTypeEnum.CHAT as ChannelTypeEnum.CHAT;
+  private axiosInstance = axios.create();
+  constructor(
+    private config: {
+      alertUid?: string;
+      title?: string;
+      imageUrl?: string;
+      state?: string;
+      externalLink?: string;
+    }
+  ) {}
+
+  async sendMessage(
+    options: IChatOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const url = new URL(options.webhookUrl);
+    const body = {
+      alert_uid: this.config.alertUid,
+      title: this.config.title,
+      image_url: this.config.imageUrl,
+      state: this.config.state,
+      link_to_upstream_details: this.config.externalLink,
+      message: options.content,
+    };
+    //response is just string "Ok."
+    const { headers } = await this.axiosInstance.post(url.toString(), body);
+
+    return {
+      id: this.config.alertUid || options.content,
```

Comment:
What does the options.content return?

Reply:
this is actually the message itself, there is no valid identifier that grafana returns in response(body/header), it it acceptable to generate a uuid and use it as it instead?

Reply:
For this use-case yes, we should not be using the actual message as the id here.

Reply:
done